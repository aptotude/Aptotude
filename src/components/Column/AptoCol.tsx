import React from 'react';
import classNames from 'classnames';
import { StandardTypes } from '../../utils/standardTypes';
import './aptoCol.scss';
import capitalize from '../../utils/capitalize';

const COMPONENT_PREFIX = 'AptoCol';

export type AptoColSize = 'auto' | 'true' | '1' | '2' | '3' | '4' | '5' | '6' | '7' | '8' | '9' | '10' | '11' | '12';

export interface AptoColDisplayProps extends StandardTypes {
  xs?: AptoColSize;
  sm?: AptoColSize;
  md?: AptoColSize;
  lg?: AptoColSize;
  xl?: AptoColSize;
  align?: 'start' | 'center' | 'end' | null | undefined;
}

export class AptoCol extends React.PureComponent<AptoColDisplayProps> {
  public render() {
    const {
      children,
      className,
      forwardRef,
      align,
      ...rest
    } = this.props;

    const classes = classNames(
      COMPONENT_PREFIX,
      this._getClass(),
      align && `${COMPONENT_PREFIX}--align${capitalize(align)}`,
      className
    );

    return (
      <div
        {...rest}
        ref={forwardRef}
        className={classes}>
        {children}
      </div>
    );
  }

  private _getClass(): string {
    const classes: string[] = [];
    if (this.props.xs) {
      classes.push(this._parseAttribute(this.props.xs, ''));
    }
    if (this.props.sm) {
      classes.push(this._parseAttribute(this.props.sm, 'sm'));
    }
    if (this.props.md) {
      classes.push(this._parseAttribute(this.props.md, 'md'));
    }
    if (this.props.lg) {
      classes.push(this._parseAttribute(this.props.lg, 'lg'));
    }
    if (this.props.xl) {
      classes.push(this._parseAttribute(this.props.xl, 'xl'));
    }
    return classes.length ? classes.join(' ') : '';
  }

  private _parseAttribute(count: any, size: string): string {
    if (size !== '') {
      size = `-${size}`;
    }
    return count === 'true' ? `${COMPONENT_PREFIX}${size}` : `${COMPONENT_PREFIX}${size}-${count}`;
  }
}
