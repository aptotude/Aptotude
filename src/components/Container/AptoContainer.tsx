import React from 'react';
import classNames from 'classnames';
import { StandardTypes } from '../../utils/standardTypes';
import './aptoContainer.scss';

const COMPONENT_PREFIX = 'AptoContainer';

export interface AptoContainerDisplayProps extends StandardTypes {
  scroll?: boolean;
  fixed?: boolean;
}

export class AptoContainer extends React.PureComponent<AptoContainerDisplayProps> {
  public static defaultProps = {
    scroll: false,
    fixed: false
  };

  public render() {
    const {
      children,
      className,
      scroll,
      fixed,
      forwardRef,
      ...rest
    } = this.props;

    const classes = classNames(
      COMPONENT_PREFIX,
      scroll && `${COMPONENT_PREFIX}--scroll`,
      fixed && `${COMPONENT_PREFIX}--fixed`,
      className
    );

    return (
      <div
        {...rest}
        ref={forwardRef}
        className={classes}>
        {children}
      </div>
    );
  }
}
