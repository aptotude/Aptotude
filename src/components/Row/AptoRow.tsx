import React from 'react';
import classNames from 'classnames';
import { StandardTypes } from '../../utils/standardTypes';
import './aptoRow.scss';
import capitalize from '../../utils/capitalize';

const COMPONENT_PREFIX = 'AptoRow';

interface Props extends StandardTypes {
  noGutter?: boolean;
  align?: 'start' | 'center' | 'end' | null | undefined;
  justify?: 'start' | 'center' | 'end' | 'around' | 'between' | null | undefined;
}

export default class AptoRow extends React.Component<Props> {
  public static defaultProps = {
    noGutter: false,
    align: null,
    justify: null
  };

  public render() {
    const {
      children,
      className,
      noGutter,
      forwardRef,
      align,
      justify,
      ...rest
    } = this.props;

    const classes = classNames(
      COMPONENT_PREFIX,
      noGutter && `${COMPONENT_PREFIX}--noGutter`,
      align && `${COMPONENT_PREFIX}--align${capitalize(align)}`,
      justify && `${COMPONENT_PREFIX}--justify${capitalize(justify)}`,
      className
    );

    return (
      <div
        {...rest}
        ref={forwardRef}
        className={classes}>
        {children}
      </div>
    );
  }
}
