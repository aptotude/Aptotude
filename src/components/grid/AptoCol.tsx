import React from 'react';
import classNames from 'classnames';
import { StandardTypes } from '../../utils/standardTypes';
import './aptoCol.scss';

const COMPONENT_PREFIX = 'AptoCol';

type RowType = 'auto' | 'true' | '1' | '2' | '3' | '4' | '5' | '6' | '7' | '8' | '9' | '10' | '11' | '12';

interface Props extends StandardTypes {
  xs?: RowType;
  sm?: RowType;
  md?: RowType;
  lg?: RowType;
  xl?: RowType;
}

class AptoCol extends React.Component<Props> {
  public render() {
    const {
      children,
      className,
      forwardRef,
      ...rest
    } = this.props;

    const classes = classNames(
      COMPONENT_PREFIX,
      this._getClass(),
      className
    );

    const Component = 'div';

    return (
      <Component
        {...rest}
        ref={forwardRef}
        className={classes}>
        {children}
      </Component>
    );
  }

  private _getClass(): string {
    const classes: string[] = [];
    if (this.props.xs) {
      classes.push(this._parseAttribute(this.props.xs, ''));
    }
    if (this.props.sm) {
      classes.push(this._parseAttribute(this.props.sm, 'sm'));
    }
    if (this.props.md) {
      classes.push(this._parseAttribute(this.props.md, 'md'));
    }
    if (this.props.lg) {
      classes.push(this._parseAttribute(this.props.lg, 'lg'));
    }
    if (this.props.xl) {
      classes.push(this._parseAttribute(this.props.xl, 'xl'));
    }
    if (classes.length) {
      return classes.join(' ');
    }
    return '';
  }

  private _parseAttribute(count: any, size: string): string {
    if (size !== '') {
      size = `-${size}`;
    }
    if (count === 'true') {
      return `${COMPONENT_PREFIX}${size}`;
    } else {
      return `${COMPONENT_PREFIX}${size}-${count}`;
    }
  }
}

export default AptoCol;