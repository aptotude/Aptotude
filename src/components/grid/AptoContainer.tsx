import React from 'react';
import classNames from 'classnames';
import { StandardTypes } from '../../utils/standardTypes';
import './aptoContainer.scss';

const COMPONENT_PREFIX = 'AptoContainer';

interface Props extends StandardTypes {
  scroll?: boolean;
  fixed?: boolean;
}

class AptoContainer extends React.Component<Props> {
  public static defaultProps = {
    scroll: false,
    fixed: false
  };

  public render() {
    const {
      children,
      className,
      scroll,
      fixed,
      forwardRef,
      ...rest
    } = this.props;

    const classes = classNames(
      COMPONENT_PREFIX,
      scroll && `${COMPONENT_PREFIX}--scroll`,
      fixed && `${COMPONENT_PREFIX}--fixed`,
      className
    );

    const Component = 'div';

    return (
      <Component
        {...rest}
        ref={forwardRef}
        className={classes}>
        {children}
      </Component>
    );
  }
}

export default AptoContainer;