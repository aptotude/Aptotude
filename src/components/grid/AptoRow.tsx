import React from 'react';
import classNames from 'classnames';
import { StandardTypes } from '../../utils/standardTypes';
import './aptoRow.scss';

const COMPONENT_PREFIX = 'AptoRow';

interface Props extends StandardTypes {
  noGutter?: boolean;
}

class AptoRow extends React.Component<Props> {
  public static defaultProps = {
    noGutter: false
  };

  public render() {
    const {
      children,
      className,
      noGutter,
      forwardRef,
      ...rest
    } = this.props;

    const classes = classNames(
      COMPONENT_PREFIX,
      noGutter && `${COMPONENT_PREFIX}--noGutter`,
      className
    );

    const Component = 'div';

    return (
      <Component
        {...rest}
        ref={forwardRef}
        className={classes}>
        {children}
      </Component>
    );
  }
}

export default AptoRow;