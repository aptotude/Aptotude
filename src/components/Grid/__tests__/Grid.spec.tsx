import React from 'react';
import renderer from 'react-test-renderer';
import AptoCol from '../../Column/AptoCol';
import AptoContainer from '../../Container/AptoContainer';
import AptoRow from '../../Row/AptoRow';

test('Grid renders', () => {
  const component = renderer.create(
    <AptoContainer>
      <AptoRow>
        <AptoCol>Foo</AptoCol>
        <AptoCol>Bar</AptoCol>
      </AptoRow>
      <AptoRow>
        <AptoCol>Baz</AptoCol>
        <AptoCol>Maz</AptoCol>
      </AptoRow>
    </AptoContainer>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
