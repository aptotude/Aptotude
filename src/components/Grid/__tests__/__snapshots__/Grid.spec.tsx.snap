// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid renders 1`] = `
<div
  className="AptoContainer"
>
  <div
    className="AptoRow"
  >
    <div
      className="AptoCol"
    >
      Foo
    </div>
    <div
      className="AptoCol"
    >
      Bar
    </div>
  </div>
  <div
    className="AptoRow"
  >
    <div
      className="AptoCol"
    >
      Baz
    </div>
    <div
      className="AptoCol"
    >
      Maz
    </div>
  </div>
</div>
`;
